cartography/intel/aws/ec2/network_acls.py

```diff
@@ -68,12 +68,12 @@ def transform_network_acl_data(
             for rule in network_acl["Entries"]:
                 direction = 'egress' if rule['Egress'] else 'inbound'
                 transformed_rule = {
-                    'Id': f"{network_acl['NetworkAclId']}/{direction}/{rule['RuleNumber']}",
-                    'CidrBlock': rule['CidrBlock'],
-                    'Egress': rule['Egress'],
-                    'Protocol': rule['Protocol'],
-                    'RuleAction': rule['RuleAction'],
-                    'RuleNumber': rule['RuleNumber'],
+                    'Id': f"{network_acl['NetworkAclId']}/{direction}/{rule.get('RuleNumber')}",
+                    'CidrBlock': rule.get('CidrBlock'),
+                    'Egress': rule.get('Egress'),
+                    'Protocol': rule.get('Protocol'),
+                    'RuleAction': rule.get('RuleAction'),
+                    'RuleNumber': rule.get('RuleNumber'),
                     # Add pointer back to the nacl to create an edge
```

Comment:
switching to `get` comes with the question: what's the default value? And what's the implication of having such broken / default data in our DB?

I think the data model for this is defined here:
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-networkaclentry.html

CIDR block is indicated as "Conditional", and it looks like either this field or `Ipv6CidrBlock` must be present. So we should update our code here to take one or another instead.

Reply:
Probably worth reviewing the rest of the parameters too.

Reply:
I wouldn't consider this data as "broken". We trust the incoming data from the AWS ACLs API, considering the optional fields. Any required transformations or enrichments can be performed ad hoc by the client consuming this data.

Reply:
I think the question here is whether cartography should provide an accurate representation of the AWS data and its constraints.

If so, then we should adhere to what @kledo-lyft and provide some kind of validation to spot the cases where we don't get any of the two values. If not, then it's fine to just store the data _as is_ and let the cartography consumers think about this.

I think the first option is better. Even though it can mean more maintenance work, consumers can be sure the data is as accurate as possible to AWS and it will also make easier to spot any bug when the data we're storing doesn't follow these constraints.

Reply:
First, sorry for the crash. It's annoying that AWS APIs are inconsistent and sometimes leave out entire fields.

>I think the data model for this is defined here:
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-networkaclentry.html

CIDR block is indicated as "Conditional", and it looks like either this field or Ipv6CidrBlock must be present. So we should update our code here to take one or another instead.

@kledo-lyft - good catch. I was referring to https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_NetworkAclEntry.html when I wrote this code. It looks like the cloudformation docs are better. 


This fix is simple:

1. do a `.get('CidrBlock')`.
2. do a `.get('Ipv6CidrBlock')`.
3. leave everything else the same (unless we know there are others where the API leaves it out).
4. update https://github.com/cartography-cncf/cartography/blob/774d67b560ea50035fe6f8ed72795ec0c3d45cf7/cartography/models/aws/ec2/network_acl_rules.py#L23 to have an ipv6cidrblock field

This works because the neo4j python driver treats Nones as if the field does not even exist, so assuming that the aws conditional is correct, our end result is that the node written to the graph will have either cidrblock or ipv6cidrblock but not both.

---

As seen here, cidrblock and ipv6cidrblock are both not id fields so it's ok for these to be None.
https://github.com/cartography-cncf/cartography/blob/774d67b560ea50035fe6f8ed72795ec0c3d45cf7/cartography/models/aws/ec2/network_acl_rules.py#L23